### DISCUSSION TITLE
Cannot open folder in OpenGrok

### DISCUSSION THREAD

**Discussion by YongJ7** [oracle/opengrok#4453] [General]
**Describe the bug**
I cannot open folder in OpenGrok,,

installed tomcat 10 & java 17(openjdk-1.17.0) & opengrok 1.12.20
http://localhost:12333 connection is OK
also http://localhost:12333/source connection is OK
But when I click dlt-daemon on opengrok,, below screenshot appears..

Below is the command I used to do indexing
```
java -Djava.util.logging.config.file=/var/opengrok/etc/logging.properties -jar /var/opengrok/dist/lib/opengrok.jar -c /usr/local/bin/ctags -s /var/opengrok/src -d /var/opengrok/data -H -P -S -G -W /var/opengrok/etc/configuration.xml -U http://localhost:12333/source
```
And I attached the log file for indexing.

![image](https://github.com/oracle/opengrok/assets/85343553/e5dd955a-215d-4098-a622-c299ba6614a3)

[opengrok0.0.log](https://github.com/oracle/opengrok/files/13111977/opengrok0.0.log)

Please help,,

**Comment by vladak** [comment]
What was the URL you accessed ? Is there anything relevant in the Tomcat logs ?

**Comment by YongJ7** [comment]
@vladak,
http://10.157.110.123:12333/source/ this is my URL for opengrok
I have attached the log file. And when I used my command in terminal to do indexing below appeared.
SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
SLF4J: Defaulting to no-operation (NOP) logger implementation
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.

**Reply by vladak** [reply]
I meant: what URL did you access that caused the error ?

**Reply by vladak** [reply]
The log file you attached is from the indexer. That looks okay. The key is what is in Tomcat logs.

**Comment by YongJ7** [comment]
@vladak,
I don't understand " what URL did I access",,
And for tomcat log,, It seems these two cause the problem
org.opengrok.indexer.util.ForbiddenSymlinkException
java.io.IOException: Could not create suggester directory

**Reply by vladak** [reply]
When the error was displayed, the URL field of the browser had a value - what was that value ? was it http://10.157.110.123:12333/source/ or something else ?

Also, the Tomcat log fragment is not very helpful, I need to see more to be able to tell what is going on.

**Comment by YongJ7** [comment]
@vladak ,
http://10.157.110.123:12333/source/xref/dlt_daemon this is the URL when error show.

And below is the log file from /opt/tomcat/logs/catalina.out

24-Oct-2023 18:53:11.725 WARNING [OpenGrok-invalidate-repos-116] org.opengrok.indexer.history.HistoryGuru.lambda$invalidateRepositories$4 Could not create git repository object for '/var/opengrok/src/dlt-daemon'
	org.opengrok.indexer.util.ForbiddenSymlinkException: /var/opengrok/src/dlt-daemon is prohibited symlink
		at org.opengrok.indexer.util.PathUtils.getRelativeToCanonical(PathUtils.java:144)
		at org.opengrok.indexer.configuration.RuntimeEnvironment.getPathRelativeToSourceRoot(RuntimeEnvironment.java:468)
		at org.opengrok.indexer.history.RepositoryFactory.getRepository(RepositoryFactory.java:190)
		at org.opengrok.indexer.history.RepositoryFactory.getRepository(RepositoryFactory.java:162)
		at org.opengrok.indexer.history.RepositoryFactory.getRepository(RepositoryFactory.java:283)
		at org.opengrok.indexer.history.HistoryGuru.lambda$invalidateRepositories$4(HistoryGuru.java:1236)
		at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
		at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
		at java.base/java.lang.Thread.run(Thread.java:829)

**Reply by vladak** [reply]
This might be related, although the error message is from repository invalidation, which happens when the webapp is deployed, not when a request is being processed.

Is `/var/opengrok/src/dlt-daemon` indeed a symlink ?

Anyhow, it is strange that such symlink was refused, given that by default all symlinks directly under source root should be allowed.

**Reply by vladak** [reply]
When the indexer is run on a source root that contains symlinks, such symlinks are stored in the configuration file specified with the `-W` indexer option, like so:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<java version="11.0.20.1" class="java.beans.XMLDecoder">
 <object class="org.opengrok.indexer.configuration.Configuration" id="Configuration0">
  <void property="allowedSymlinks">
   <void method="add">
    <string>/var/opengrok/src.symlinks/openssl</string>
   </void>
  </void>
  <void property="dataRoot">
   <string>/var/opengrok/data.symlinks</string>
...
```

 When the webapp starts, it will read this configuration and the list of allowed symlinks will become part of the runtime configuration. What is in your `/var/opengrok/etc/configuration.xml` ?

**Comment by YongJ7** [comment]
@vladak ,
No matter what I did, it didn't work, so I reinstalled Ubuntu 18.04 and tried again, and it worked normally.
Let's close this issue. Anyway, thank you for your help
